### DISCUSSION TITLE
Is there a way to configure NATS servers in an automated manner?

### DISCUSSION THREAD

**Discussion by vweckerle** [nats-nui/nui#37] [Q&A]
Hi all,

I am wondering whether there is a any way how one could configure NATS servers with username+password auth in a NATS NUI Docker installation via either environment variables or files in mounted filesystems?

I'd like to automate the configuration for a specific NATS server in NATS NUI without having to do anything via the UI.

Greetings
Valentin

**Comment by pricelessrabbit** [comment]
Hi, 
currently is not directly supported, but it will probably be in the near future. There's also a opened issue about it (#27 ).
However, with the v0.5.0 release, you can put in a directory the connections you need with the format of NATS CLI contexts (same as the json saved by the nats cli when creating a context) and from UI you can import all the contexts in the directory. Now is a manual action but my plan is  to add also some sort of config flag so that when the application start, it auto imports all the contexts files found in the directory passed in the flag.
For now as a workaround you can:
- put all the connection in a dir with the natsl cli format and at the first deploy, manually importing them from the UI
- use the [rest api](https://github.com/nats-nui/nui/blob/main/frontend/docs/entities/connection/import.md) in some way to automate the import: you share the dir with contexts and then you call the api every time you need to update the imports. it will import the new records and update the already imported ones.
